integration/tests/upgrades/version_beacon_service_event_test.go

```diff
@@ -126,6 +163,83 @@ func (s *TestServiceEventVersionControl) TestEmittingVersionBeaconServiceEvent()
 		s.Require().Equal(patch, uint8(version.Patch))
 	})
 
+	s.Run("stop with version beacon", func() {
+		latestFinalized, err := s.AccessClient().GetLatestFinalizedBlockHeader(ctx)
+		require.NoError(s.T(), err)
+
+		// make sure target height is correct
+		// the height at which the version change will take effect should be after
+		// the current height + the freeze period
+		height := latestFinalized.Height + freezePeriod + 100
```

Comment:
I feel like we could use a buffer a bit less than 100 here (20?). We only need to wait for the collection to be incorporated into a block.